### PR TITLE
Changes to launch AEA app from Broker's WebView

### PR DESCRIPTION
**Pull request by shahzaibj**
This is needed to unblock our partner. Broker will launch the AEA app when it receives the above redirect with scheme aea://  (essentially by launching an Intent for this URI). 

At this stage, the AEA app will come to foreground and user can complete their action in the app.

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java [hunk 1]
```diff
@@ -359,6 +367,14 @@ private void launchCompanyPortal() {
         returnResult(RawAuthorizationResult.ResultCode.MDM_FLOW);
     }
 
+    private void processAmazonAppUri(@NonNull final String url) {
```
Comment (iamgusain) [inline common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java:392]: Do we need to explicitly stop/close the webview after this, or that is something that is already happening.

Comment (shahzaibj) [reply common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java:392]: No. The webview needs to keep running. The user returns back to the flow after completing action in AEA app.

**Review by iamgusain — APPROVED** [review]

File: common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java (hunk 1 above)
Comment (iamgusain) [reply common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java:392]: I see, so the web view is showing a page that indirectly waits for user performing an action in AEA app.

Comment (shahzaibj) [reply common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java:392]: Yes, correct.

File: common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java [hunk 2]
```diff
@@ -178,6 +179,9 @@ private boolean handleUrl(final WebView view, final String url) {
             } else if (isAuthAppMFAUrl(formattedURL)) {
                 Logger.info(methodTag,"Request to link account with Authenticator.");
                 processAuthAppMFAUrl(url);
+            } else if (isAmazonAppRedirect(url)) {
```
Comment (melissaahn) [inline common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java]: isAmazonAppRedirect(formattedURL)?

**Review by melissaahn — APPROVED** [review]

File: common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java (hunk 2 above)
Comment (shahzaibj) [reply common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java]: Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2264/commits/b7d387cb12b8a6255f8c95d15d282cd8f0be98a4

**Pull request merged**